src/lib.rs

```diff
@@ -281,36 +329,26 @@ pub fn run_tests<D>(
     // Print number of tests
     printer.print_title(tests.len() as u64);
 
-    // Execute all tests
     let mut failed_tests = Vec::new();
     let mut num_ignored = 0;
     let mut num_benches = 0;
     let mut num_passed = 0;
-    for test in &tests {
+    // Execute all tests
+    for (test, outcome) in run_tests_threaded(args, tests, run_test) {
+        // Print `test foo    ... ok` etc.
+        printer.print_test(&test.name, &test.kind);
+        printer.print_single_outcome(&outcome);
```

Comment:
This is not exactly how libtest works. The test name is printed before it is executed if test threads are 1. So I guess just add an `if` condition here and add a `print_test` with the inverted condition in the `run_tests_threaded` at the appropriate  place?

Reply:
The problem is that would require passing the `printer` to the threads, which is not allowed because it's not Sync

Reply:
Actually, there's an alternative: threads could send through the queue event both when they start and finish running a given test. I'll do that.